shared/src/types/vote_extensions/validator_set_update.rs

```diff
@@ -370,4 +387,89 @@ mod tests {
+            hot_key_addr: EthAddress([0; 20]),
+            cold_key_addr: EthAddress([0; 20]),
+        };
+        let validator_b = EthAddrBook {
+            hot_key_addr: EthAddress([1; 20]),
+            cold_key_addr: EthAddress([1; 20]),
+        };
+
+        assert_eq!(
+            compare_voting_powers_map_items(
+                &(&validator_a, &same_voting_power),
+                &(&validator_b, &same_voting_power),
+            ),
+            Ordering::Less
+        );
+    }
+
+    /// Checks that comparing two [`VotingPowersMap`] items with different
+    /// voting powers results in the item with the lesser voting power being
+    /// regarded as "greater".
+    #[test]
+    fn test_compare_voting_powers_map_items_different_voting_powers() {
+        let validator_a = EthAddrBook {
+            hot_key_addr: EthAddress([0; 20]),
+            cold_key_addr: EthAddress([0; 20]),
+        };
+        let validator_a_voting_power = 200.into();
+        let validator_b = EthAddrBook {
+            hot_key_addr: EthAddress([1; 20]),
+            cold_key_addr: EthAddress([1; 20]),
+        };
+        let validator_b_voting_power = 100.into();
+
+        assert_eq!(
+            compare_voting_powers_map_items(
+                &(&validator_a, &validator_a_voting_power),
+                &(&validator_b, &validator_b_voting_power),
+            ),
+            Ordering::Less
+        );
+    }
+
+    /// Checks that [`VotingPowersMapExt::get_abi_encoded`] gives a
+    /// deterministic result in the case where there are multiple validators
+    /// with the same voting power.
+    ///
+    /// NB: this test may pass even if the implementation is not
+    /// deterministic unless the test is run with the `--release` profile, as it
+    /// is implicitly relying on how iterating over a [`HashMap`] seems to
+    /// return items in the order in which they were inserted, at least for this
+    /// very small 2-item example.
+    #[test]
+    fn test_voting_powers_map_get_abi_encoded_deterministic_with_identical_voting_powers()
+     {
+        let validator_a = EthAddrBook {
+            hot_key_addr: EthAddress([0; 20]),
+            cold_key_addr: EthAddress([0; 20]),
+        };
+        let validator_b = EthAddrBook {
+            hot_key_addr: EthAddress([1; 20]),
+            cold_key_addr: EthAddress([1; 20]),
+        };
+        let same_voting_power = 200.into();
+
+        let mut voting_powers_1 = VotingPowersMap::default();
+        voting_powers_1.insert(validator_a.clone(), same_voting_power);
+        voting_powers_1.insert(validator_b.clone(), same_voting_power);
+
+        let mut voting_powers_2 = VotingPowersMap::default();
+        voting_powers_2.insert(validator_b, same_voting_power);
+        voting_powers_2.insert(validator_a, same_voting_power);
+
+        let x = voting_powers_1.get_abi_encoded();
+        let y = voting_powers_2.get_abi_encoded();
+        assert_eq!(x, y);
+    }
```

Comment:
Having this test that would only fail in `--release` mode if broken is not a great solution. I think we should split out `get_abi_encoded` into a fn that works with `&mut [(&EthAddrBook, &VotingPower)]` directly, then we can easily test that with vectors directly, rather than relying on `HashMap::iter`